### PR TITLE
[+] MongoDB HasMany - Allow documents embedded to an array to be editable

### PR DESCRIPTION
**Pull request by ArnaudValensi**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by ArnaudValensi** [comment]
[(3) Embedded HasMany - Friendo wants to edit documents in array of embedded documents](https://trello.com/c/mh9laEoU/2265-3-embedded-hasmany-friendo-wants-to-edit-documents-in-array-of-embedded-documents)

**Comment by ArnaudValensi** [comment]
[(3) Embedded HasMany - Friendo wants to edit documents in array of embedded documents](https://trello.com/c/mh9laEoU/2265-3-embedded-hasmany-friendo-wants-to-edit-documents-in-array-of-embedded-documents)

File: routes/resources.js [hunk 1]
```diff
@@ -110,6 +123,8 @@ module.exports = function (app, model, Implementation, integrator, opts) {
       auth.ensureAuthenticated, this.get);
     app.post(path.generate(modelName, opts), auth.ensureAuthenticated,
       this.create);
+    app.put(path.generate(modelName + '/:recordId/relationships/:recordIndex', opts),
+      auth.ensureAuthenticated, this.updateEmbedded);
```
[the remark this answers is not in the document]
Comment (ArnaudValensi) [reply routes/resources.js]: Done

File: routes/associations.js [hunk 2]
```diff
@@ -113,6 +114,22 @@ module.exports = function (app, model, Implementation, integrator, opts) {
       .catch(next);
   }
 
+  function updateEmbeddedDocument(association) {
+    return function (request, response, next) {
+      new ResourceDeserializer(Implementation, model, request.body, false)
```
Comment (arnaudbesnier) [inline routes/associations.js]: Should be returned I think.

**Pull request merged**